services/manage/app/components/user/modal-form.js

```diff
@@ -1,21 +1,58 @@
 import Component from '@ember/component';
+import ActionMixin from '@base-cms/id-me-manage/mixins/action-mixin';
+import { ObjectQueryManager } from 'ember-apollo-client';
+import gql from 'graphql-tag';
 import { inject } from '@ember/service';
+import { computed } from '@ember/object';
 
-export default Component.extend({
-  router: inject(),
+const query = gql`
+  query UserProfile {
+    activeUser {
+      id
+      givenName
+      familyName
+    }
+  }
+`;
 
+const mutation = gql`
+  mutation UserProfileEdit($input: UpdateUserProfileMutationInput!) {
+    updateUserProfile(input: $input) {
+      id
+    }
+  }
+`;
+
+export default Component.extend(ActionMixin, ObjectQueryManager, {
+  user: inject(),
   tagName: '',
   title: null,
   isActionRunning: false,
+  isModalCloseable: computed.reads('user.profileCloseable'),
 
-  init() {
+  async init() {
     this._super(...arguments);
-    if (!this.model) this.set('model', {});
+    this.set('model', await this.apollo.watchQuery({ query }, 'activeUser'));
   },
 
   actions: {
-    returnToList() {
-      return this.router.transitionTo('manage.orgs.list');
+    async update(closeModal) {
+      try {
+        this.startAction();
+        const { givenName, familyName, photoURL } = this.get('model');
+        const input = { givenName, familyName, photoURL };
+        const variables = { input };
+        await this.apollo.mutate({ mutation, variables }, 'updateUserProfile');
+        this.set('isModalOpen', false);
```

Comment:
I'm slightly confused by this... the `closeModal()` function should already set this property, assuming it's bound to `@show`